File: Khong_chuyen/C++/ma_hoa.cpp
// https://oj.vnoi.info/problem/olp_kc21_encryption

#include <bits/stdc++.h>
using namespace std;

int main()
{
    ios::sync_with_stdio(false); 
    cin.tie(0); 
    cout.tie(0);
    long l,r,a,k;
    cin >> l >> r >> a >> k;
    long h = __gcd(a,k) , x=k/h , dem=0;
    long tmp=x;
    for (auto i=1 ; tmp<=r ; i++ , tmp=x*i)
    {
        if (x*i>=l) dem++;
    }
    cout << dem;
    return 0;
}

File: Khong_chuyen/Python/ma_hoa.py
# https://oj.vnoi.info/problem/olp_kc21_encryption

from math import gcd

l,r,a,k=map(int,input().split())
h=gcd(a,k)
x=k//h
dem=0
i=1
temp=x
while temp<=r:
	if x*i>=l:
		dem+=1
	i+=1
	temp=x*i
print(dem)
